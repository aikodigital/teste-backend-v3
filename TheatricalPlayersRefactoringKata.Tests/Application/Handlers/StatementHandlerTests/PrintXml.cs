using ApprovalTests;
using ApprovalTests.Reporters;
using TheatricalPlayers.Application.Handlers;
using TheatricalPlayers.Core.Entities;
using TheatricalPlayers.Core.Enums;
using TheatricalPlayers.Core.Interfaces.Statements;
using TheatricalPlayersRefactoringKata.Tests.Application.Mocks;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests.Application.Handlers.StatementTests;

public class PrintXml
{
    private readonly IStatementPrinterHandler _statementPrinterHandler; 
    public PrintXml()
    {
        _statementPrinterHandler = new StatementPrinterHandler();
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void PrintXmlOk()
    {
       
        var (plays, invoice) = InvoiceSampleDataGenerator.CreateSampleData();

        var xml = _statementPrinterHandler.PrintXml(invoice, plays);
        
        Approvals.Verify(xml);
    }
}