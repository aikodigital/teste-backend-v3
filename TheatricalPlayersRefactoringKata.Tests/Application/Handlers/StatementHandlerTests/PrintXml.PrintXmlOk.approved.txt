<?xml version="1.0" encoding="utf-8"?>
<Statement xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Customer>BigCo</Customer>
  <Items>
    <Item>
      <Play>Hamlet</Play>
      <AmountOwed>650</AmountOwed>
      <EarnedCredits>25</EarnedCredits>
      <Seats>55</Seats>
    </Item>
    <Item>
      <Play>As You Like It</Play>
      <AmountOwed>547</AmountOwed>
      <EarnedCredits>12</EarnedCredits>
      <Seats>35</Seats>
    </Item>
    <Item>
      <Play>Othello</Play>
      <AmountOwed>456</AmountOwed>
      <EarnedCredits>10</EarnedCredits>
      <Seats>40</Seats>
    </Item>
    <Item>
      <Play>Henry V</Play>
      <AmountOwed>705.4</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>20</Seats>
    </Item>
    <Item>
      <Play>john</Play>
      <AmountOwed>931.6</AmountOwed>
      <EarnedCredits>9</EarnedCredits>
      <Seats>39</Seats>
    </Item>
    <Item>
      <Play>Richard III</Play>
      <AmountOwed>803.6</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>20</Seats>
    </Item>
  </Items>
  <AmountOwed>4093.6</AmountOwed>
  <EarnedCredits>56</EarnedCredits>
</Statement>